### PR TITLE
add error raise to LH token service

### PR DESCRIPTION
**Pull request by aherzberg**
## Summary
LH Token service for mobile immunizations was assuming the service would return a 200. This raises an error based on what the response is. the example that we saw in datadog was 429 too many requests error. This may be an error we want to pass along to FE but rather then creating this one off mapping, I created this ticket, https://app.zenhub.com/workspaces/va-mobile-60f1a34998bc75000f2a489f/issues/gh/department-of-veterans-affairs/va-mobile-app/8959, to create one place to map all these types of errors so it can be used across the API.  

## Related issue(s)
https://github.com/department-of-veterans-affairs/va-mobile-app/issues/8522

## Testing done

- [ ] *New code is covered by unit tests*
- *Describe what the old behavior was prior to the change*
- *Describe the steps required to verify your changes are working as expected. Exclusively stating 'Specs run' is NOT acceptable as appropriate testing*
- *If this work is behind a flipper:*
  - *Tests need to be written for both the flipper on and flipper off scenarios. [Docs](https://depo-platform-documentation.scrollhelp.site/developer-docs/feature-toggles-guide#Featuretogglesguide-Backendexample).*
  - *What is the testing plan for rolling out the feature?*

## Screenshots
_Note: Optional_

## What areas of the site does it impact?
*(Describe what parts of the site are impacted and*if*code touched other areas)*

## Acceptance criteria

- [ ]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [ ]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

**Review by kpethtel — CHANGES_REQUESTED** [review]
This looks good except the test description.

FWIW, I think the reason we have to do this is because fetching the lighthouse token is a tag-along in the client and not a first class citizen. `Common::Client::Base` establishes a pattern of using the `perform` method to make faraday requests. But the perform method relies on the included config's `connection` method. Since the token request uses `access_token_connection`, you can't use the `perform` method, which means you don't get the error handling from the `Common::Client::Base` `request` method. You can see that in this same lighthouse service, we use `perform` for the non-token requests, which is why we don't have to rescue and re-raise them. 

In other words, an alternative way of fixing this is to split the token fetch into its own service with its own configuration. Then we would get all of the built-in error handling. That's not a big deal and I don't think it's worth additional work here to change it. It just means that in this case, any error we receive while making that token request will result in a 502, regardless of whether it's a timeout, unparseable response, or whatever. Which is fine IMO.

File: modules/mobile/spec/request/v1/immunizations_request_spec.rb [hunk 1]
```diff
@@ -112,6 +112,24 @@
...
 
+    context 'when token service returns an error' do
+      before do
+        VCR.use_cassette('mobile/lighthouse_health/get_immunizations_token_too_many_error',
+                         match_requests_on: %i[method uri]) do
+          get '/mobile/v1/health/immunizations', headers: sis_headers, params: { page: { size: 12, number: 1 } }
+        end
+      end
+
+      it 'returns a 500' do
```
[the remark this answers is not in the document]
Comment (aherzberg) [reply modules/mobile/spec/request/v1/immunizations_request_spec.rb]: yeah, it originally returned a 500 before I rescued and re raised it. forgot to update that part. Agreed its probably not worth the effort of putting in its own service but that makes sense on why this needs special logic.

**Review by kpethtel — APPROVED** [review]
👍

File: modules/mobile/spec/request/v1/immunizations_request_spec.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (kpethtel) [reply modules/mobile/spec/request/v1/immunizations_request_spec.rb]: Yeah, I only mention it because I had to think through, "Why do we have to do it this way?"
